keycloak/themes/hmda/login/register.ftl

```diff
@@ -96,4 +96,104 @@
             </div>
         </form>
     </#if>
+    <script>
+        var institutionSearchUri = "${properties.institutionSearchUri!}/institutions";
+
+        function emailToDomain(email) {
+            return email.split("@", 2)[1];
+        }
+
+        function getFormEmail() {
+            return $("#email").val().trim().toLowerCase();
+        }
+
+        function isValidDomain(email, domain) {
+            return emailToDomain(email) === domain;
+        }
+
+        function getStatusIcon(email, domain) {
+            var statusIcon = '';
+            if(isValidDomain(email, domain))
+                statusIcon =  '<i style="color:#20aa3f;" class="fa fa-check-circle" aria-hidden="true"></i>';
+            else
+                statusIcon =  '<i style="color:#ff9e1b;" class="fa fa-warning" aria-hidden="true"></i>';
+
+            return statusIcon;
+        }
+
+        $(document).ready(function() {
+            
+            $("#user\\.attributes\\.institutions").select2({
+                placeholder: "Select Institution(s)",
+                minimumInputLength: 3,
+                multiple: true,
+                allowClear: true,
+                width: "600px",
+                dropdownCssClass: "bigdrop",
+                ajax: {
+                    url: institutionSearchUri,
+                    data: function(term, page) {
+                        // Search based on user input
+                        return { search: term }
+
+                        // Search based on "email" form field
+                        //var domain = emailToDomain($("#email").val());
+                        //return { domain: domain }
+                    },
+                    results: function(data, page) {
+                        return {
+                            // Each result MUST have an `id` attribute
+                            results: data.results 
+                        }
+                    }
+                },
+                escapeMarkup: function(markup) {
+                    return markup;
+                },
+                formatSelection: function(institution) {
+                    return  institution.name + ' (' + institution.id + ') ' + getStatusIcon(getFormEmail(), institution.domain[0]);
+                },
+                formatResult: function(institution) {
```

Comment:
If ever there was an HTML string crying out to be templated ;)

I don't really want this to be changed (especially because I'd prefer not to have a front-end build step for the keycloak themes), but I will not fail to marvel at it.